src/test/java/staffconnect/logic/commands/DeleteMeetingCommandTest.java

```diff
@@ -0,0 +1,166 @@
+
+        assertCommandSuccess(deleteMeetingCommand, TEST_MODEL, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidPersonIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(TEST_MODEL.getFilteredPersonList().size() + 1);
+        DeleteMeetingCommand deleteMeetingCommand = new DeleteMeetingCommand(outOfBoundIndex, INDEX_FIRST_MEETING);
+
+        assertCommandFailure(deleteMeetingCommand, TEST_MODEL, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidMeetingIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(TEST_MODEL
+                .getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased())
+                .getFilteredMeetings().size() + 1);
+
+        DeleteMeetingCommand deleteMeetingCommand = new DeleteMeetingCommand(INDEX_FIRST_PERSON, outOfBoundIndex);
+
+        assertCommandFailure(deleteMeetingCommand, TEST_MODEL, Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        Person personToSelect = copyPersonWithNewMeetingManager(TEST_MODEL.getFilteredPersonList()
+                .get(INDEX_FIRST_PERSON.getZeroBased()));
+        personToSelect.addMeetings(VALID_MEETING); //First valid meeting
+        personToSelect.addMeetings(VALID_MEETING_STUDY); //Second valid meeting
+        showMeetingAtIndex(personToSelect, INDEX_SECOND_MEETING); //Update visibility
+        //update the model
+        TEST_MODEL.setPerson(personToSelect, personToSelect);
+
+        Meeting meetingToDelete = personToSelect.getFilteredMeetings().get(INDEX_FIRST_MEETING.getZeroBased());
+        //deletes the second meeting in the model
+        DeleteMeetingCommand deleteMeetingCommand = new DeleteMeetingCommand(INDEX_FIRST_PERSON, INDEX_FIRST_MEETING);
+        String expectedMessage = String.format(DeleteMeetingCommand.MESSAGE_DELETE_MEETING_SUCCESS,
+                Messages.format(meetingToDelete));
+
+        Model expectedModel = new ModelManager(getTypicalStaffBook(), new UserPrefs());
+        Person expectedPersonSelect =
+                copyPersonWithNewMeetingManager(expectedModel.getFilteredPersonList()
+                        .get(INDEX_FIRST_PERSON.getZeroBased()));
+
+        expectedPersonSelect.addMeetings(VALID_MEETING); //This should be the only meeting left in the model
+        showNoMeeting(expectedPersonSelect); //But it is not visible
+        expectedModel.setPerson(expectedPersonSelect, expectedPersonSelect);
+
+        assertCommandSuccess(deleteMeetingCommand, TEST_MODEL, expectedMessage, expectedModel);
+    }
+
+    /**
+     * Updates {@code person}'s filtered list to show no meeting.
+     */
+
+    private void showNoMeeting(Person person) {
+        person.updateFilteredMeetingList(p -> false);
+        assertTrue(person.getFilteredMeetings().isEmpty());
+    }
```

Comment:
could you move the `showNoMeeting()` to the top to separate methods for test utility and `@Test` methods

Reply:
Shifted!